packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -103,6 +104,23 @@ export interface PipelineProps {
    * @default - None.
    */
   readonly stages?: StageProps[];
+
+  /**
+   * Create KMS keys for cross-account deployments
+   *
+   * This controls whether the pipeline is enabled for cross-account deployments.
+   *
+   * By default cross-account deployments are enabled, but this feature requires
+   * that KMS Customer Master Keys are created which have a cost of $1/month.
+   *
+   * If you do not need cross-account deployments, you can set this to `false` to
+   * not create those keys and save on that cost (the artifact bucket will be
+   * encrypted with an AWS-managed key). However, cross-account deployments will
+   * no longer be possible.
+   *
+   * @default true
+   */
+  readonly crossAccountKeys?: boolean;
```

Comment:
Should we be smarter in our default here? If cross-account actions are defined, pick true to be the default, and false otherwise?

Reply:
MMmmmmaybe. The problem starts happening when you start flipping between `true` and `false`. 

I'd rather have made it opt-in but that's not something we can do before v2 :(

Reply:
> Creating/deleting/creating keys is not something I want to do without explicit user control.

Curious why not? We do this for other kinds of resources. 
Is it because content that was encrypted with a key will effectively become inaccessible when the key is deleted?

Reply:
That is one reason, and there are cost implications as well (in fact that's the whole point of this PR).

Can I put in an `// @deprecated(v2): switch to default false` ?

Reply:
I suppose, yes.